cypress/e2e/trigger_builder.cy.js

```diff
@@ -81,33 +81,53 @@ const checkTriggerEditorUIValues = (trigger) => {
         .contains('Any interface')
         .should('be.selected');
     } else {
-      cy.get('#triggerInterfaceName')
-        .scrollIntoView()
-        .should('be.visible')
-        .contains(iface)
-        .should('be.selected');
-      cy.get('#triggerInterfaceMajor')
+      cy.wait(2000);
+      cy.window().then((win) => {
+        if (win.monaco && win.monaco.editor) {
+          const editor = win.monaco.editor.getModels()[0];
+          const editorContent = editor.getValue();
+          const editorJson = JSON.parse(editorContent);
+
+          expect(editorJson.simple_triggers[0].interface_name).to.equal(iface);
+        } else {
+          throw new Error('Monaco Editor instance is not available');
+        }
+      });
+      cy.window().then((win) => {
+        if (win.monaco && win.monaco.editor) {
+          const editor = win.monaco.editor.getModels()[0];
+          const editorContent = editor.getValue();
+          const editorJson = JSON.parse(editorContent);
+
+          expect(editorJson.simple_triggers[0].interface_major).to.equal(simpleTrigger.interface_major);
+        } else {
+          throw new Error('Monaco Editor instance is not available');
+        }
+      });
+      cy.window().then((win) => {
+        if (win.monaco && win.monaco.editor) {
+          const editor = win.monaco.editor.getModels()[0];
+          const editorContent = editor.getValue();
+          const editorJson = JSON.parse(editorContent);
+
+          expect(editorJson.simple_triggers[0].match_path).to.equal(simpleTrigger.match_path);
+        } else {
+          throw new Error('Monaco Editor instance is not available');
+        }
```

Comment:
Here there is a repetition of 
```
      cy.window().then((win) => {
        if (win.monaco && win.monaco.editor) {
          const editor = win.monaco.editor.getModels()[0];
          const editorContent = editor.getValue();
          const editorJson = JSON.parse(editorContent);
```
all of that can be merged in a single if clause